frontend/model/notifications/periodicNotifications.js

```diff
@@ -38,59 +34,147 @@ const validateNotificationData = objectOf({
   shouldClearStateKey: isFunction
 })
 
-async function runNotificationListRecursive (data) {
+/**
+ * Runs a recursive notification list handler that checks and emits
+ * notifications based on specified conditions and delays.
+ *
+ * The function performs the following steps:
+ * 1. Checks if enough time has passed since the last run based on the specified
+ * delay.
+ * 2. Iterates over each notification entry in the `data.notifications` array:
+ *    - If the notification has not been fired and its emit condition evaluates
+ *      to true, it:
+ *        - Calls the emit function associated with the notification.
+ *        - Marks the notification as fired in the `firedMap`.
+ *    - If the notification has been fired and its clear condition evaluates to
+ *      true, it:
+ *        - Removes the notification from the `firedMap`.
+ * 3. Updates the `lastRun` timestamp to the current time.
+ * 4. Sets a timeout to recursively call `runNotificationListRecursive` after
+ * the specified delay, adjusting for the time that has already passed since the
+ * last run.
+ */
+async function runNotificationListRecursive () {
   const rootState = sbp('state/vuex/state')
   const rootGetters = sbp('state/vuex/getters')
-  const firedMap = rootState.periodicNotificationAlreadyFiredMap
-  const callWithStates = func => func.call(data.state, { rootState, rootGetters })
+  const firedMap = rootState.periodicNotificationAlreadyFiredMap.alreadyFired
+  const lastRunMap = rootState.periodicNotificationAlreadyFiredMap.lastRun
+  const callWithStates = (func, stateKey) => func.call(ephemeralNotificationState.partition[stateKey], { rootState, rootGetters })
 
-  for (const entry of data.notifications) {
-    if (!firedMap[entry.stateKey] && callWithStates(entry.emitCondition)) {
-      await callWithStates(entry.emit)
-      Vue.set(firedMap, entry.stateKey, true)
-    }
+  // Exit if a timeout is already set
+  if (ephemeralNotificationState.clearTimeout) return
+  // Prevent accidentally calling `runNotificationListRecursive` while it's
+  // still running
+  // We set `clearTimeout` to a local value to stop processing notifications
+  // when clearing stop notifications, to avoid calling
+  // `runNotificationListRecursive` while it's running and to be able to compare
+  // it to the local value later to replace `clearTimeout` later on in this
+  // function
+  let aborted: boolean = false
+  const noop = () => { aborted = true }
```

Comment:
Is this really a `noop`? It's clearly doing something.

Reply:
Good point.